Formula/zim.rb

```diff
@@ -0,0 +1,47 @@
+class Zim < Formula
+  desc "Graphical text editor used to maintain a collection of wiki pages"
+  homepage "https://zim-wiki.org/"
+  url "https://github.com/jaap-karssenberg/zim-desktop-wiki/archive/0.72.1.tar.gz"
+  sha256 "ba02e418b4fb1d7847f96b49ada8c917c881a28bb5fb55dcdca54be7b3fd196a"
+  head "https://github.com/jaap-karssenberg/zim-desktop-wiki.git"
+
+  bottle do
+    cellar :any_skip_relocation
+    sha256 "91f15636c51398dc31f7c43d570a652cc1bddb8fc3571bdfc203d25231aa095d" => :catalina
+    sha256 "91f15636c51398dc31f7c43d570a652cc1bddb8fc3571bdfc203d25231aa095d" => :mojave
+    sha256 "91f15636c51398dc31f7c43d570a652cc1bddb8fc3571bdfc203d25231aa095d" => :high_sierra
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "adwaita-icon-theme"
+  depends_on "graphviz"
+  depends_on "gtk+3"
+  depends_on "gtksourceview3"
+  depends_on "pygobject3"
+  depends_on "python"
```

Comment:
```suggestion
  depends_on "python@3.8"
```

Reply:
Could you explain why it is better to require a non-default version than the default python provided by homebrew?

Reply:
Python 3.8 is provided by homebrew. It's just versioned while all formula are being migrated to 3.8.

Reply:
> Python 3.8 is provided by homebrew.

Sure, I was asking about the default (unversioned) python.

> It's just versioned while all formula are being migrated to 3.8.

But not all the dependencies are available for python@3.8. Making the requested changes (and also changing `system 'python'` to `system 'python3.8'` fails with:
```
==> python3.8 ./setup.py install --prefix=/usr/local/Cellar/zim/0.72.1/libexec
Traceback (most recent call last):
  File "./setup.py", line 25, in <module>
    import makeman # helper script
  File "/private/tmp/zim-20200302-48231-1aye8gu/zim-desktop-wiki-0.72.1/makeman.py", line 13, in <module>
    from zim.main import HelpCommand
  File "/private/tmp/zim-20200302-48231-1aye8gu/zim-desktop-wiki-0.72.1/zim/main/__init__.py", line 22, in <module>
    import zim.fs
  File "/private/tmp/zim-20200302-48231-1aye8gu/zim-desktop-wiki-0.72.1/zim/fs.py", line 27, in <module>
    from zim.signals import SignalEmitter, SIGNAL_AFTER
  File "/private/tmp/zim-20200302-48231-1aye8gu/zim-desktop-wiki-0.72.1/zim/signals.py", line 7, in <module>
    from gi.repository import GObject
ModuleNotFoundError: No module named 'gi'
```

So, I have left out this, and applied the rest of the suggestions.